src/pfe/portal/modules/ExtensionList.js

```diff
@@ -14,16 +14,68 @@ const path = require('path');
 const Extension = require('./Extension.js');
 const Logger = require('./utils/Logger');
 const ExtensionListError = require('./utils/errors/ExtensionListError');
+const utils = require('./utils/sharedFunctions');
+const { promisify } = require('util');
+const exec = promisify(require('child_process').exec);
 
 const log = new Logger(__filename);
 
+const extensionsDir = '/extensions';
+const extensionsPattern = /^(\S+)(-[\d.]+)\.zip$/; // e.g. extension-name-0.0.1.zip
+const suffixOld = '__old';
+
 /**
  * The ExtensionList class
  */
 module.exports = class ExtensionList {
   constructor() {
     this._list = {};
   }
+  
+  /**
+   * Install (unzip) built-in extensions that are stored in /extensions to the
+   * given target directory
+   * 
+   * @param {string} targetDir, the target directory to install extensions to 
+   */
+  async installBuiltInExtensions(targetDir) {
+  
+    // get the zips from the /extensions directory
+    const entries = await fs.readdir(extensionsDir, { withFileTypes: true });
+    
+    for (let entry of entries) {
+      
+        let match;
+  
+        // look for files with names matching the expected pattern
+        if (entry.isFile() && (match = extensionsPattern.exec(entry.name))) {
```

Comment:
Comment: I think debugging would be easier if we separated this assignment and if statement, but https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec suggests that this is a common pattern with regex

Reply:
I'll keep it as-is then, in the interest of keeping the code looking cleaner.